### ISSUE TITLE
clarify documentation of `Pair::from_seed`

### ISSUE THREAD

**Issue by gnunicorn** [paritytech/substrate#5755]
The comments around `from_seed_slice` specify that this is a dangerous method only to be used for testing, however the upper of `Pair` uses it without that information passed on and the client is using the same functionality to generate the tests keys for `Alice`, `Bob`, etc.

We should clarify the documentation to clearly stated under which circumstances this features is to be expected to be used safely.

**Comment by dud1337** [comment]
What would it take to get this done?

**Comment by bkchr** [comment]
This issue + the linked one are rather fishy issues. I don't understand the problem? As described in the trait description, you should only use these functions when you have a secure seed. If you are in doubt, you should not use them. This is already told by the docs. These functions exists, as said by @gnunicorn for some testing purposes where you don't care that the seed is secure which is totally fine. I don't think that this needs more clarification.

TLDR: If the seed is insecure, your private key will be insecure. The docs say that you only should use this when you are sure that the seed is secure. Not sure what else you expect as documentation for such a function.

**Comment by dud1337** [comment]
I'm good with closing this based on the above.
